src/components/Editor/index.js

```diff
@@ -201,24 +202,38 @@ class Editor extends PureComponent {
     // `this.props` to be the current props. This lifecycle method is
     // responsible for updating the editor annotations.
     const { selectedLocation, selectedSource } = this.props;
-
+    if (!selectedLocation || !selectedLocation.location) {
+      return;
+    }
     // If the location is different and a new line is requested,
     // update the pending jump line. Note that if jumping to a line in
     // a source where the text hasn't been loaded yet, we will set the
     // line here but not jump until rendering the actual source.
-    if (prevProps.selectedLocation !== selectedLocation) {
-      if (selectedLocation && selectedLocation.line != undefined) {
+
+    if (
+      !prevProps.selectedLocation ||
+      prevProps.selectedLocation.location !== selectedLocation.location
+    ) {
+      if (selectedLocation.location.line != undefined) {
         this.pendingJumpLocation = selectedLocation;
       } else {
         this.pendingJumpLocation = null;
       }
     }
 
+    // (i.e the line and column are different).
     // Only update and jump around in real source texts. This will
     // keep the jump state around until the real source text is
     // loaded.
-    if (selectedSource && selectedSource.has("text")) {
-      this.flashLine();
+    if (
+      !prevProps.selectedLocation ||
+      !prevProps.selectedLocation.location ||
+      (prevProps.selectedLocation &&
+        prevProps.selectedLocation.location &&
```

Comment:
why do we now care about `prevProps.selectedLocation.location`?

Reply:
good point!